CHANGELOG.md

```diff
@@ -17,6 +17,8 @@ Changelog for `ta4j`, roughly following [keepachangelog.com](http://keepachangel
 - :tada: **Enhancement** Added possibility to use CostModels when backtesting with the BacktestExecutor
 - :tada: **Enhancement** added Num#zero, Num#one, Num#hundred
 - **Example** added a json serialization and deserialization example of BarSeries using google-gson library
+- :tada: **Enhancement** Implemented an (abstract) TradeBasedIndicator, which can perform calculations based on the last trade
+- :tada: **Enhancement** Implemented a BreakEvenIndicator, which returns the needed break even regarding the last trade according to given transaction fees
```

Comment:
> 
> 
> I have reservation against this indicator type. Similar to #761 why should we mix up `Trades` and `indicators`? Wouldn't a `Rule` be the place for such a decisions? In a `Rule` you have access to the `TradingRecord` of the current run. Having access to a `TradingRecord` in an `Indicator` means that you have to create another `Strategy` with `Rules` and `Indicators` and so on just to apply this `TradingRecord` to this indicator. There may be special cases for this, but I am not sure if ta4j should support this by default and within the current architecture of this framework

Mhm, not sure if that can be handled in Rules. If you have any ideas how to do it, I'm open to every suggestion.

I use the trade-based indicator quite much in my trading bot.
Examples:
* Break even indicator
    * --> if break even is reached, the exit criterion of the strategy is changed (needs only true and false)
    * --> place an order 2% above the break even (needs the concrete value/Num of the break even for every trade. Thats why it is an "calculator"(Indicator) and not a "condition"(Rule))
* Lowest/highest since last sell
* Restart of rules/conditions after a sell
    * --> example: Buy if EMA(20)>EMA(200). Sell on 2% gain. Without a "reset", the bot would directly buy again if the EMA(20) is still over EMA(200).
* BuyPrice-Indicator --> Indicator which holds the buy price during the lifetime of a trade
* IsBuyPhase indicator ..> true if in BuyPhase and false if not
* ...

Instead of writing an indicator, I could write different strategies and call my bot on different strategies according to the current state of the bot. But then, I cannot perform backtests with taa4j anymore, because multiple strategies cannot be glued together for the backtests.